hydromt/models/components/grid.py

```diff
@@ -13,371 +11,163 @@
 from pyproj import CRS
 from shapely.geometry import box
 
-from hydromt import workflows
-from hydromt.gis import raster, utils
-from hydromt.models.api import Model, parse_region
-
-__all__ = ["GridModel"]
-logger = logging.getLogger(__name__)
-
-
-class GridMixin(object):
-    # placeholders
-    # xr.Dataset representation of all static parameter maps at the same resolution and
-    # bounds - renamed from staticmaps
-    _API = {"grid": xr.Dataset}
+from hydromt._typing.error import NoDataStrategy, _exec_nodata_strat
+from hydromt._typing.type_def import DeferedFileClose
+from hydromt.data_catalog import DataCatalog
+from hydromt.gis import raster
+from hydromt.gis import utils as gis_utils
+from hydromt.io.readers import read_nc
+from hydromt.io.writers import write_nc
+from hydromt.models.components.model_component import ModelComponent
+from hydromt.models.root import ModelRoot
+from hydromt.workflows.basin_mask import get_basin_geometry, parse_region
+from hydromt.workflows.grid import (
+    grid_from_constant,
+    grid_from_geodataframe,
+    grid_from_raster_reclass,
+    grid_from_rasterdataset,
+    rotated_grid,
+)
+
+logger = getLogger(__name__)
+
+
+class GridComponent(ModelComponent):
```

Comment:
I think it's good that the import names actually reflect the hirarchy. Users can just import them from the model sub module because we'll export them through the `__init__.py` so for them it would still be 
`from hydromt.models import GridComponent` or even `from hydromt.model.components import GridComponent` if they want to be more specific

Reply:
Let's discuss this in the next harmonisation PR